### ISSUE TITLE
Alexa stops building cxx object SampleApp ... UIManager.app (97%)

### ISSUE THREAD

**Issue by GeekTeacher** [alexa/avs-device-sdk#1429]
**IMPORTANT**: Before you create an issue, please take a look at our [Issue Reporting Guide](https://github.com/alexa/avs-device-sdk/wiki/Issue-Reporting-Guide).

### What version of the AVS Device SDK are you using?
      <x.y.z>

### Tell us what hardware you're using:
- [ ] Desktop / Laptop
- [x] Raspberry Pi
- [ ] Other - tell us more:

### Tell us about your OS (Type & version):
- [ ] Linux
- [ ] MacOS 
- [x] Raspbian Stretch
- [ ] Raspbian Jessy
- [ ] Other - tell us more:

**Comment by mvelegon-amzn** [comment]
This issue sometimes happens because of limited resources on the pi. Do you have any other application running on the pi. You could also modify the shell script to reduce the number of jobs.

**Comment by GeekTeacher** [comment]
Many thanks for your reply. My laptop computer ran out of power whilst send the original message and so I hadn’t completed it and I didn’t actually think it had been sent. 

I am running a Raspberry Pi 3 off a 64gb SD card with a fresh build of the latest version of Raspian Stretch. I tried using Raspian Duster but got error messages during the initial updates so went back to Stretch. With that in mind I can’t see how it should be running out of resources with such a large card and only the basic build with no extra apps.

I note that there are several versions of how to implement Alexa on a Pi (including one on the Amazon Developer site) and was wondering if you could suggest the most reliable instructions to try again. I have tried several and many times.

I have got as far as the install recognising the Alexa wake up message but then got a message to do with the network not being connected and working in ‘limited mode’. The microphone and playback audio work fine.

Any help such as a modified script or other suggestions would be much appreciated.

Cheers, Martyn.

**Comment by dhpp** [comment]
Hi @GeekTeacher, it sounds like you basically have the SDK compiled & working, however the client entered 'limited' mode since Authorization failed.  I would suggest that you remove the database files (as listed in the AlexaClientSDKConfig.json file) and go through the authorization steps again.

**Comment by celinval** [comment]
Hi @GeekTeacher, were you able to try @dhpp suggestion? Did it fix the problem? You can also use the 'Factory Reset' option from the limited mode (key 'k') to clean up all the database files and start again.

**Comment by kclchan** [comment]
I am closing this issue due to inactivity. Please feel free to re-open it if it has been closed in error.

**Comment by monforum** [comment]
**The problem comes from** :
Low power supply > when CPU Load is high > CPU drains more power > Raspberry reboots, so the installation is interrupted at this level. 🥇 

**Solution**: plug 2 Amps power supply (instead of computer USB plug suplying), then redo the installation, the high CPU load will

**Comment by zorofrizzy** [comment]
I had the same issue as @GeekTeacher , I identified the problem with reference to @monforum  and solved it using the following steps : 

I have a screen and I get the HDMI output over there, I connect a mouse and a Keyboard too.

### **TLDR - I just turned off bluetooth and VNC, rebooted to Commandline Interface (Non GUI and build it again)**

```
**Solution:**
1. Turn off Bluetooth in the Pi, you can use these commands ( I just right clicked and turned off the icon) 


$ echo "dtoverlay=pi3-disable-bt" | sudo tee -a /boot/config.txt
$ sudo systemctl disable hciuart


2. Reboot to Command Line Interface (CLI)
         a. $ sudo raspi-config
         b. Go to boot options 
         c. Go to "Enable Boot to Desktop/Scratch
         d. Select Console / Text Autologin (CLI)

3. Reboot the system and re run the build script as earlier. 
```

**Suggestion** - As you re run the script, try to remove your USB mouse and Keyboard so that they don't consume power. Also using SSH instead of VNC should save you more power in case you don't have a screen available. Turning down VNC is also recommended.

```
Mentions

- Raspberry Pi 3B+
- Raspbian Buster March 2020 Build
- VNC Enabled
- 5V, 2A Power Source used for pi
- 100mA usage based both Keyboard and Mouse connected (200 mA total)
```


> 
> 
> **The problem comes from** :
> Low power supply > when CPU Load is high > CPU drains more power > Raspberry reboots, so the installation is interrupted at this level. 🥇
> 
> **Solution**:
> 
>     1. unplug all USB devices (mic, speaker...etc) form your Raspberry to remove any power suply drainer
> 
>     2. plug 2 Amps power supply (instead of computer USB plug suplying)
> 
>     3. redo the installation, the high CPU load will

**Comment by joselcuevam** [comment]
increasing swap size solved for me, change it from 100 to 1024 re execute it finishes at this time, changes at power supply  did not make difference for me

**Comment by TinkerJeff** [comment]
I had a similar issue, the steps I followed to expand the swapfile on the sdcard were:
sudo dphys-swapfile swapoff
sudo nano /etc/dphys-swapfile
_Increase the number 100 to 1100__
sudo dphys-swapfile setup
sudo dphys-swapfile swapon

Use swapon -s to view the swap space.

I elected to restart afterwards to verify the change worked after a restart.
